cmd/kube-apiserver/app/server.go

```diff
@@ -355,13 +368,22 @@ func BuildGenericConfig(s *options.ServerRunOptions, proxyTransport *http.Transp
 	if err := s.GenericServerRunOptions.ApplyTo(genericConfig); err != nil {
 		return nil, nil, nil, nil, nil, err
 	}
+
 	insecureServingOptions, err := s.InsecureServing.ApplyTo(genericConfig)
 	if err != nil {
 		return nil, nil, nil, nil, nil, err
 	}
+	if insecureServingOptions != nil {
+		ln, err := CreateListener(insecureServingOptions.BindAddress)
+		if err != nil {
+			return nil, nil, nil, nil, nil, err
+		}
+		insecureServingOptions.Listener = ln
```

Comment:
Do we actually have cases where we need this mechanism at all for the insecure port? We are trying to phase out the insecure listening. That's why we moved the InsecureServingOptions to kubeapiserver. If possible, I would not add this feature here all.

Reply:
>We are trying to phase out the insecure listening.
I know, But insecure server now also use `RunServer()` as secure one. So can not run insecure server if not init listener here.

Reply:
But all this logic should go to `s.InsecureServing.ApplyTo`.